examples/apps/src/ble_bas_peripheral.rs

```diff
@@ -24,7 +24,10 @@ struct Server {
 // Battery service
 #[gatt_service(uuid = "180f")]
 struct BatteryService {
-    #[characteristic(uuid = "2a19", read, write, notify, on_read = battery_level_on_read, on_write = battery_level_on_write)]
+    /// Battery Level
+    #[descriptor(uuid = "2b20", read, write, value = "Battery Level", capacity = 100, on_read = battery_level_on_read)]
+    #[descriptor(uuid = "2b21", read, write, value = "Other Descriptor", capacity = 255, on_write = battery_level_on_write)]
+    #[characteristic(uuid = "2a19", read, write, notify, on_read = battery_level_on_read, on_write = battery_level_on_write, value = 10)]
```

Comment:
I'm thinking it's relying too much on macros here, and that maybe this is a signal it should be designed differently. What if one could define descriptors like this:

```
#[characteristic(uuid = "2a19", read, write, notify, on_read = battery_level_on_read, on_write = battery_level_on_write, value = 10)]
level: u8,

#[descriptor(uuid = "2b21", read, write, value = "Other Descriptor")
level_descriptor: Vec<u8, 255>,

```

What's missing in this example is how to tie the descriptor to the characteristic. Maybe one could represent it like this:

```
#[characteristic(uuid = "2a19)]
level: Characteristic {
    #[value(read, write, notify, on_read = battery_level_on_read, on_write = battery_level_on_write, value = 10)]
    value: u8,
    #[descriptor(uuid = "2b21", read, write, value = "Other Descriptor")
    descriptor: Vec<u8, 255>,
}
```

Reply:
Hmm, I don't know about these alternatives, I feel they add more complexity to the simple case (no descriptors) or decouple the characteristic and descriptor.

We could reduce some visual noise by shortening the names of the inputs? i.e. 'read' becomes 'r' etc.

@petekubiak made a good point though that we don't currently have a unique name per descriptor for the get and set methods, so that's still a challenge to make ergonomic.

Reply:
I think the simple case could still be as is today. This more complex form can be used only if complex descriptors are needed.

Shortening the name of the inputs would make the code harder to read, so would be against that.

With the above, it should be simple to generate appropriate names for setting and getting the descriptor values, since the macro can use the descriptor field name.

Reply:
At first I thought that the struct variant would require a whole new proc-macro to deal with the characteristic structure, but it's just occurred to me that we can parse the input token stream however we want, so we can use the existing `#[gatt_service]` macro to parse the Characteristic `structure` and flatten the generated fields into the service struct.
I think this is a good solution in that it allows us to both associate the descriptors with their characteristics and ergonomically provide a name for the descriptor for use with `get()` and `set()`.